packages/eslint-plugin-react-hooks/__tests__/ESLintRuleExhaustiveDeps-test.js

```diff
@@ -7000,22 +6997,122 @@ const testsTypescript = {
       errors: [
         {
           message:
-            "React Hook useEffect has missing dependencies: 'pizza.crust' and 'pizza.toppings'. " +
+            "React Hook useEffect has missing dependencies: 'pizza.crust' and 'pizza?.toppings'. " +
             'Either include them or remove the dependency array.',
           suggestions: [
             {
-              // TODO the description and suggestions should probably also
-              // preserve the optional chaining.
               desc:
-                'Update the dependencies array to be: [pizza.crust, pizza.toppings]',
+                'Update the dependencies array to be: [pizza.crust, pizza?.toppings]',
```

Comment:
Note: technically this could be further improved to not add `?` at all because another dep didn't use it. However, I don't think it matters because at least this doesn't cause a crash. Additionally, this points your attention to unnecessary `?` inside the function — since presumably if data is immutable, you shouldn't need it anyway.